azblob/parsing_urls.go

```diff
@@ -124,6 +134,11 @@ func (up BlobURLParts) URL() url.URL {
 
 	rawQuery := up.UnparsedParams
 
+	// Check: Both snapshot and version id cannot be present in the request URL.
+	if up.Snapshot != "" && up.VersionID != "" {
+		errors.New("Snapshot and versioning cannot be enabled simultaneously")
```

Comment:
>errors.New("Snapshot and versioning cannot be enabled simultaneously") [](start = 2, length = 70)

This is not getting returned though? you can also avoid this check and simply let the service give back an error message

Reply:
Ohh understood. I'll remove.  Since blob versioning and snapshot cannot be used at the same time I was thinking of putting a check in our code base.